apollo-federation/src/connectors/json_selection/location.rs

```diff
@@ -19,10 +20,33 @@ use super::ParseResult;
 // parsing and then only set Some(message) when we need to report an error, so
 // we would not be cloning long String messages very often (and the rest of the
 // Span fields are cheap to clone).
-pub(crate) type Span<'a> = LocatedSpan<&'a str, Option<&'static str>>;
+pub(crate) type Span<'a> = LocatedSpan<&'a str, SpanExtra>;
+
+#[derive(Debug, PartialEq, Eq, Clone, Hash, Default)]
+pub(crate) struct SpanExtra {
+    pub(super) spec: ConnectSpec,
+    pub(super) errors: Vec<String>,
+}
 
 pub(super) fn new_span(input: &str) -> Span {
-    Span::new_extra(input, None)
+    Span::new_extra(input, Default::default())
+}
```

Comment:
Here are the changes that require parser input `Span`s to be cloned in more cases. Note that the `&'a str` reference is copied over when we clone a `Span`, so it's only really the `extra: SpanExtra` stuff that requires `Clone`.